ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Stages/FcsErrorsStageProcessBase.fs

```diff
@@ -68,6 +69,17 @@ module FSharpErrors =
     let [<Literal>] ifExprMissingElseBranch = "This 'if' expression is missing an 'else' branch."
     let [<Literal>] expressionIsAFunctionMessage = "This expression is a function value, i.e. is missing arguments. Its type is "
 
+    let private unexpectedTypeError1 = Regex("This expression (was) expected to have type\n    '(.+)'    \nbut here has type\n    '(.+)'")
+    let private unexpectedTypeError2 = Regex("All elements of (an array|a list) must be of the same type as the first element, which here is '(.+)'. This element has type '(.+)'.")
+
+    let tryGetUnexpectedTypeErrorInfo message =
```

Comment:
Could we try using an active pattern here? :)
http://www.fssnip.net/29/title/Regular-expression-active-pattern